[#57816] create hierarchy items

Kharonus opened the pull request:
# Ticket
[OP#57816](https://community.openproject.org/work_packages/57816)

- added form for item creation
- added borderbox in items component

# What are you trying to accomplish?
- enable creation of hierarchy items on top most level

# What approach did you choose and why?
- enable turbo stream updates to render inline form to create items
- add form validation
- render existing items in borderbox

Kharonus committed:
[#57816] create hierarchy items

- https://community.openproject.org/work_packages/57816
- added form for item creation
- added borderbox in items component

Kharonus committed:
[#57816] added turbo streams to update the forms

- render hierarchical item in component
- add missing routes
- localize dry-validation error

Kharonus committed:
[#57816] final polishing

- removing ineffective system arguments
- add i18n support for validation messages

akabiru commented on app/components/admin/custom_fields/hierarchy/items_component.rb:
```diff
@@ -0,0 +1,53 @@
+        end
+
+        def items
+          # TODO: This must be context aware (breadcrumbs)
```
🟢 open TODO

Kharonus replied:
Ah, yes, that's a todo for the feature, but not for the implementation. There is a follow up implementation ticket, that inserts logic in there dependent on the the current hierarchy level.

akabiru reviewed (approved):
🥇

Kharonus committed:
Merge branch 'dev' into implementation/57816-enable-item-creation

Kharonus committed:
[#57816] fixed contract unit test

Kharonus committed:
[#57816] disabled lazy load i18n for eager loaded runs

Kharonus committed:
[#57816] added contract test with localized error validation

cbliard committed:
Use native I18n implementation of lazy loading of i18n data

cbliard committed:
Make LocaleExtractor recognize our js-<locale>.yaml file format

mereghost committed:
hacks the hacky hacksaw

cbliard commented on spec/support/i18n_lazy_loading.rb:
```diff
@@ -28,16 +28,19 @@
+
+    return if locale.nil?
+
+    locale.tr("js-", "").to_sym
```
`tr` is cryptic and a lot less intention-revealing than other methods like `delete_prefix` or `gsub`.
And also it does not work as expected as it deletes every `j`, `s`, and `-` characters instead of deleting the word `js-`

```
>> "js-ja".tr("js-", "")
=> "a"
>> "js-ja".delete("js-")
=> "a"
>> "js-ja".delete_prefix("js-")
=> "ja"
```

Better use `delete_prefix`.

cbliard committed:
Fix extraction of locale from translation files

The `LocaleExtractor` recognizes these kind of files:

  - `<locale>.yaml`
  - `<locale>_something.yaml`

It does not recognize these files we use:

  - `js-<locale>.yaml` (locale is extracted as `js-<locale>`)
  - `<locale>.seeders.yaml` (locale is extracted as `<locale>.seeders`)

It's fixed by deleting the `js-` prefix and the `.seeders` suffix from
the extracted locale.

Kharonus committed:
[#57816] disabled locale spec temporarily
